### PR TITLE
Added Testcases to scoring_functions.py

### PR DESCRIPTION
**Pull request by cozek**
This PR is in response to #1085 asking for doctests. I have added doctests to all functions in
scoring_functions.py

### CODE REVIEW

**Comment by cclauss** [comment]
Conflicting files and failing tests.

File: arithmetic_analysis/bisection.py [hunk 1]
```diff
@@ -1,21 +1,25 @@
 import math
 
 
-def bisection(function, a, b):  # finds where the function becomes 0 in [a,b] using bolzano
+def bisection(
+    function, a, b
+):  # finds where the function becomes 0 in [a,b] using bolzano
```
Comment (cclauss) [inline arithmetic_analysis/bisection.py]: Please convert this comment to a docstring.

File: arithmetic_analysis/bisection.py [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
     start = a
     end = b
     if function(a) == 0:  # one of the a or b is a root for the function
         return a
     elif function(b) == 0:
         return b
-    elif function(a) * function(b) > 0:  # if none of these are root and they are both positive or negative,
+    elif (
+        function(a) * function(b) > 0
+    ):  # if none of these are root and they are both positive or negative,
```
Comment (cclauss) [inline arithmetic_analysis/bisection.py]: Please put this comment on a separate line.

File: arithmetic_analysis/intersection.py [hunk 3]
```diff
@@ -1,17 +1,24 @@
 import math
 
-def intersection(function,x0,x1): #function is the f we want to find its root and x0 and x1 are two random starting points
+
+def intersection(
+    function, x0, x1
+):  # function is the f we want to find its root and x0 and x1 are two random starting points
```
Comment (cclauss) [inline arithmetic_analysis/intersection.py]: Please convert this comment to a docstring.

**Comment by cozek** [comment]
This PR is insane. Something is wrong. I only changed 1 file. What the frick!?

**Comment by cozek** [comment]
Closing this.

**Pull request closed without merging**